Chaincase.UI/ViewModels/ReceiveViewModel.cs

```diff
@@ -36,6 +47,45 @@ public void InitNextReceiveKey()
         public HdPubKey ReceivePubKey { get; set; }
 
         public string BitcoinUri => $"bitcoin:{Address}";
+        public string P2EPUri => $"bitcoin:{Address}?pj={_P2EPServer.PaymentEndpoint}";
+
+        public void GenerateP2EP(string password)
+        {
+            if (!_P2EPServer.HiddenServiceIsOn)
+            {
+                StartPayjoin(password);
+            }
+            P2EPAddress = _P2EPServer.PaymentEndpoint;
+        }
+
+        public void StartPayjoin(string password)
+        {
+            var cts = new CancellationToken();
+            _P2EPServer.StartAsync(cts);
+            _P2EPServer.Password = password;
+            Logger.LogInfo($"P2EP Server listening created: {_P2EPServer.PaymentEndpoint}");
+        }
+
+        public async Task TryStartPayjoin(string password)
+        {
+            IsBusy = true;
+            string walletFilePath = Path.Combine(_walletManager.WalletDirectories.WalletsDir, $"{_network}.json");
+            try
+            {
+                await Task.Run(() => KeyManager.FromFile(walletFilePath).GetMasterExtKey(password ?? ""));
```

Comment:
dam we have to know about _walletManager & KeyManager here? looks like _wM should just expose a utility